### PR TITLE
Improve boolean indexing docs

### LINKED ISSUE #42384: DOC: Where is the "&" "|" "^" operators of Index docs?
I can't find the docs about this.

image

### PR DESCRIPTION
**Pull request by madisongoggin**
- [ X] closes #42384
- [ X] tests added / passed
- [ X] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ X] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.4.0.rst [hunk 1]
```diff
@@ -32,7 +32,7 @@ Other enhancements
 - :meth:`Series.sample`, :meth:`DataFrame.sample`, and :meth:`.GroupBy.sample` now accept a ``np.random.Generator`` as input to ``random_state``. A generator will be more performant, especially with ``replace=False`` (:issue:`38100`)
 -  Additional options added to :meth:`.Styler.bar` to control alignment and display (:issue:`26070`)
 - :meth:`Series.ewm`, :meth:`DataFrame.ewm`, now support a ``method`` argument with a ``'table'`` option that performs the windowing operation over an entire :class:`DataFrame`. See :ref:`Window Overview <window.overview>` for performance and functional benefits (:issue:`42273`)
--
+- The docs for Boolean vector indexing were enhanced (:issue:`42384`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.4.0.rst:35]: Don't need a whatsnew entry for doc improvements

File: doc/source/user_guide/indexing.rst [hunk 2]
```diff
@@ -881,6 +894,7 @@ Using a boolean vector to index a Series works exactly as in a NumPy ndarray:
    s
    s[s > 0]
    s[(s < -1) | (s > 0.5)]
+   s[(s < -1) ^ (s > 0.5)]
```
Comment (mroeschke) [inline doc/source/user_guide/indexing.rst:897]: Could you add an example for `&`

File: doc/source/user_guide/indexing.rst [hunk 3]
```diff
@@ -867,7 +867,20 @@ Boolean indexing
 .. _indexing.boolean:
 
 Another common operation is the use of boolean vectors to filter the data.
-The operators are: ``|`` for ``or``, ``&`` for ``and``, and ``~`` for ``not``.
+The operators are: ``|`` for ``or``,  ``^`` for ``bitwise or``, ``&`` for ``and``, and ``~`` for ``not``.
```
Comment (mroeschke) [inline doc/source/user_guide/indexing.rst:870]: This sentence seem redundant now with the table you provided (which is a lot better IMO)

Comment (madisonestabrook) [reply doc/source/user_guide/indexing.rst]: I reverted back to before my whatsnew commit. Then, I added and committed an example for  `&` and  removed the redundant sentence.

Sorry about the whatsnew! I saw that in the PR template and thought - for whatever reason - it was required.

File: doc/source/user_guide/indexing.rst [hunk 4]
```diff
@@ -866,8 +866,15 @@ Boolean indexing
 
 .. _indexing.boolean:
 
-Another common operation is the use of boolean vectors to filter the data.
-The operators are: ``|`` for ``or``, ``&`` for ``and``, and ``~`` for ``not``.
+Another common operation is the use of boolean vectors to filter the data. Below is a table containing available boolean vectors.
+
+| Boolean Vector | Description   |
```
Comment (mzeitlin11) [inline doc/source/user_guide/indexing.rst:871]: Should this be operator instead of Boolean Vector? Same in the sentence above.

**Comment by mroeschke** [comment]
Thanks for the PR, but it appears to have gone stale. This was close, but if you could push some small updates based on the review we would be happy to accept this PR. Closing in the meantime.

**Pull request closed without merging**